genai-perf/genai_perf/config/generate/search_parameters.py

```diff
@@ -39,25 +39,34 @@ class SearchParameters:
         "concurrency",
         "request_rate",
     ]
-    linear_range_parameters = ["instance_count"]
+
+    linear_range_parameters = ["instance_count", "num_prompts"]
 
     model_parameters = [
         "model_batch_size",
         "instance_count",
         "max_queue_delay",
     ]
-    runtime_parameters = ["runtime_batch_size", "concurrency", "request_rate"]
 
-    all_parameters = model_parameters + runtime_parameters
+    runtime_pa_parameters = ["runtime_batch_size", "concurrency", "request_rate"]
+
+    runtime_gap_parameters = [
+        "num_prompts",
+    ]
+
+    all_parameters = model_parameters + runtime_pa_parameters + runtime_gap_parameters
 
     def __init__(
         self,
-        config: ConfigCommand,
+        config: ConfigOptimize,
         is_bls_model: bool = False,
         is_ensemble_model: bool = False,
         is_composing_model: bool = False,
     ):
-        self._config = config.optimize
+        self._config = config
+        self._subcommand = (
```

Comment:
Minor nit: it feels like this could be an `Enum` instead of a string.

Reply:
No, this is a good suggestion. I've gone ahead and implemented this.